### PR TITLE
Cloneptr feature

### PR DESCRIPTION
**Pull request by hamishun**
- added cloneptr free function that takes any pointer type and clones that pointer if the object pointed to has a clone method.
- tests added
- etc

### CODE REVIEW

**Comment by hamishun** [comment]
Jenkins run tests

File: include/helib/ClonedPtr.h [hunk 1]
```diff
@@ -29,6 +29,22 @@
 
 namespace helib {
 
+/**
+ * @brief Clone any type of pointer.
+ * @param ptr Pointer to be cloned. That is the object that it points to is
+ *`cloned`.
```
Comment (jlhcrawford) [inline include/helib/ClonedPtr.h:35]:
```suggestion
 * `cloned`.
```

**Review by jlhcrawford — CHANGES_REQUESTED** [review]
Just a missing space in the comment.

**Review by jlhcrawford — APPROVED** [review]
Looks good to me. Can be merged, if tests pass.

**Pull request merged**